docs/admission-control-secure.md

```diff
@@ -0,0 +1,340 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: opa
+  name: configmap-modifier
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["update", "patch"]
+---
+# Grant OPA/kube-mgmt role defined above.
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: opa
+  name: opa-configmap-modifier
+roleRef:
+  kind: Role
+  name: configmap-modifier
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: Group
+  name: system:serviceaccounts:opa
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: opa
+  namespace: opa
+spec:
+  selector:
+    app: opa
+  ports:
+  - name: https
+    protocol: TCP
+    port: 443
+    targetPort: 443
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: opa
+  namespace: opa
+  name: opa
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: opa
+  template:
+    metadata:
+      labels:
+        app: opa
+      name: opa
+    spec:
+      containers:
+        - name: opa
+          image: openpolicyagent/opa:0.10.0
+          args:
+            - "run"
+            - "--server"
+            - "--tls-cert-file=/certs/tls.crt"
+            - "--tls-private-key-file=/certs/tls.key"
+            - "--addr=0.0.0.0:443"
+            - "--addr=http://127.0.0.1:8181"
+            - "--authentication=token"
+            - "--authorization=basic"
+            - "/policies/inject.rego" # authorization policy used on startup
+            - "--ignore=.*"           # exclude hidden dirs created by Kubernetes
+          volumeMounts:
+            - readOnly: true
+              mountPath: /certs
+              name: opa-server
+            - readOnly: true
+              mountPath: /policies
+              name: inject-policy
+        - name: kube-mgmt
+          image: openpolicyagent/kube-mgmt:0.8-dev
+          args:
+            - "--replicate-cluster=v1/namespaces"
+            - "--replicate=extensions/v1beta1/ingresses"
+            - "--opa-auth-token=kube-mgmt"
+            - "--require-policy-label=true"
+      volumes:
+        - name: opa-server
+          secret:
+            secretName: opa-server
+        - name: inject-policy
+          configMap:
+            name: inject-policy
+---
+kind: ConfigMap
+apiVersion: v1
+
+metadata:
+  name: opa-default-system-main
+  namespace: opa
+  labels:
+    openpolicyagent.org/policy: rego
+data:
+  main: |
+    package system
+
+    import data.kubernetes.admission
+
+    main = {
+      "apiVersion": "admission.k8s.io/v1beta1",
+      "kind": "AdmissionReview",
+      "response": response,
+    }
+
+    default response = {"allowed": true}
+
+    response = {
+        "allowed": false,
+        "status": {
+            "reason": reason,
+        },
+    } {
+        reason = concat(", ", admission.deny)
+        reason != ""
+    }
+---
+kind: ConfigMap
```

Comment:
I think it would be useful to highlight this configmap on it's own, i.e., instead of duplicating all of the manifests here, just show this one.

Reply:
Done